src/Utility/FieldFunctions.h

```diff
@@ -281,11 +293,18 @@ class VecFieldFunction : public VecFunc, private FieldsFuncBase
   virtual ~VecFieldFunction() {}
 
   //! \brief Sets the active patch.
-  virtual bool initPatch(size_t pIdx) { return this->setPatch(pIdx); }
+  bool initPatch(size_t pIdx) override { return this->setPatch(pIdx); }
 
 protected:
   //! \brief Evaluates the vectorial field function.
-  virtual Vec3 evaluate(const Vec3& X) const;
+  Vec3 evaluate(const Vec3& X) const override;
+
+  //! \brief Returns the gradient of the function as a 1D array.
+  std::vector<Real> evalGradient(const Vec3& X) const override
+  { return this->getGradient(X); }
+  //! \brief Returns the hessian of the function as a 1D array.
+  std::vector<Real> evalHessian(const Vec3& X) const override
+  { return this->getHessian(X); }
```

Comment:
Do we need this extra layer in the evaluation methods, or can' we just implement the bodies of the get[Values|Gradient|Hessian] methods directly in the respective eval... methods?

Reply:
needs access to field which are private in the subclass.

Reply:
I suspected there was a reason, I just did not see it. Accepted.